core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -41,6 +41,11 @@ private CatalogProperties() {}
 
   public static final boolean CACHE_ENABLED_DEFAULT = true;
 
+  /** Controls whether the caching catalog will cache table entries using case-sensitive keys. */
+  public static final String CACHE_CASE_SENSITIVE = "cache.case-sensitive";
+
+  public static final boolean CACHE_CASE_SENSITIVE_DEFAULT = false;
```

Comment:
I see the current wrapCatalog two-arg call defaults to caseSensitive=true?

Reply:
Yes, and that is the problem. I'd argue that we should default the case sensitivity to false, because I think this causes the least surprise to users.
If others have use cases where they need case sensitivity to be true, I'd like to hear about them. I may yet be convinced that case sensitivity should default to true.
In any case, this critical point is that now the case sensitivity can be configured.

Reply:
Should we match the spark sql case sensitivity patan?

Reply:
@RussellSpitzer are you referring to https://github.com/apache/spark/blob/v3.4.0/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L867-L873
That conf defaults to false, so I am in fact matching it.

Reply:
Hm I'm not too familiar, but is tthis a breaking change then?  If the user set case-sensitive=true on spark-sql side, and is running with expectation for Iceberg to follow that?  I guess @RussellSpitzer is suggesting we read the spark conf and set that from SparkCatalog?

Reply:
Exactly, reading the parameter from the runtime conf creating the catalog and using that

Reply:
Not that we have to do that, but I think that would make sense

Reply:
@RussellSpitzer I had not understood what you meant.
@szehon-ho I have now updated the PR.

Assuming caching is enabled, the behavior is as follows:
If the user configures `cache.case-sensitive`, the conf is used. Otherwise, the case sensitivity of `CachingCatalog` is set to match `spark.sql.caseSensitive` in the runtime configuration of the `SparkSession`.
Thus we have configurability as well as a default that should match users' expectations. In practice, `spark.sql.caseSensitive` should rarely be set (and thus default to false), as the Spark documentation for it advises, "It is highly discouraged to turn on case sensitive mode."

Reply:
In case there are other use cases I do not know about, I am leaving configurability in. Also, other catalogs such as `FlinkCatalog` that use the `CachingCatalog` may choose to make use of the `cache.case-sensitive` conf.

Reply:
Great, makes sense to me.